paf-mvp-operator-express/tests/integration-tests/operator.integration.write.test.ts

```diff
@@ -0,0 +1,341 @@
+    body: JSON.stringify(sampleIdsAndPreferences),
+  });
+  const proxyPostIdsPrefsResponse = await operatorClient.getWriteResponse(request);
+  proxyPostIdsPrefsResponse.url = proxyPostIdsPrefsResponse.url.replace(/^https?:\/\/[^/]+/i, '');
+  return proxyPostIdsPrefsResponse;
+};
+const getRedirectWriteUrl = async (operatorClient: OperatorClient, returnUrl = defaultReturnUrl) => {
+  const request = new MockExpressRequest({
+    headers: {
+      referer: refererUrl,
+    },
+    query: {
+      returnUrl: returnUrl,
+      message: JSON.stringify(sampleIdsAndPreferences),
+    },
+  });
+  const postIdsPrefsUrl = await operatorClient.getWriteRedirectResponse(request);
+  return postIdsPrefsUrl.replace(/^https?:\/\/[^/]+/i, '');
+};
+const publicKeyProvider = (host: string) => {
+  if (host === ClientBuilder.defaultHost) return Promise.resolve(ClientBuilder.defaultPublicKey);
+
+  throw new UnableToIdentifySignerError(`Error calling Identity endpoint on ${host}`);
+};
+describe('write', () => {
+  const getContext = async (
+    validator: IJsonValidator = JsonValidator.default(),
+    specificPublicKeyProvider = publicKeyProvider
+  ) => {
+    const operator = OperatorUtils.buildOperator(validator, specificPublicKeyProvider);
+
+    const startMock = jest.spyOn(operator, 'beginHandling');
+    const endMock = jest.spyOn(operator, 'endHandling');
+    await operator.setup();
+
+    const server: Express = operator.app.expressApp;
+
+    return { server, operator, startMock, endMock };
+  };
+  const cases = [
+    {
+      name: 'rest',
+      isRedirect: false,
+    },
+    {
+      name: 'redirect',
+      isRedirect: true,
+    },
+  ];
+  describe.each(cases)('$name', (input) => {
+    const isRedirect = input.isRedirect;
+    const getRequest = isRedirect ? getRedirectWriteUrl : getRestWriteRequest;
+    const assertError = isRedirect ? assertRedirectError : assertRestError;
+    let response = undefined;
+    it('should fallback to unknown error in case of an exception', async () => {
+      // Note that the operator node is not start()ed
+      const exceptionValidator = {
+        start: jest.fn(),
+        validate: () => {
+          throw 'UnknownException';
+        },
+      };
+
+      const { server, startMock, endMock } = await getContext(exceptionValidator);
+
+      const operatorClient = new ClientBuilder().setClientHost('no-permission.com').build(publicKeyProvider);
+      if (isRedirect) {
+        response = await supertest(server)
+          .get(await getRedirectWriteUrl(operatorClient))
+          .set('referer', refererUrl)
+          .set('Origin', refererUrl);
+      } else {
+        const writeRequest = await getRestWriteRequest(operatorClient);
+        response = await supertest(server)
+          .post(writeRequest.url)
+          .type('text/plain')
+          .send(JSON.stringify(writeRequest.payload));
+      }
+
+      assertError(response, 500, NodeErrorType.UNKNOWN_ERROR);
+      if (isRedirect) {
```

Comment:
For consistency, can you please either:
- use `verifyRedirectUrl`
- or update the "read" tests to match this pattern (I know we have discussed it in the past and I agree this version is probably more readable)

Thanks